src/main/java/com/ezasm/gui/WindowCloseListener.java

```diff
@@ -0,0 +1,26 @@
+package com.ezasm.gui;
+
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+
+import static com.ezasm.gui.menubar.MenuActions.*;
+import static com.ezasm.gui.util.DialogFactory.promptYesNoDialog;
+
+public class WindowCloseListener extends WindowAdapter {
+
+    public WindowCloseListener() {
+        super();
+    }
+
+    @Override
+    public void windowClosing(WindowEvent e) {
+        if (!Window.getInstance().getEditor().getFileSaved()) {
+            boolean resp = promptYesNoDialog("Exiting...",
+                    "Your changes have not been saved.\nWould you like to save them?");
```

Comment:
This isn't fully platform independent. Not all platforms use `\n` for newline. To get a system dependent newline character use `System.getProperties("line.separator")`